core/src/main/java/io/questdb/TelemetryJob.java

```diff
@@ -0,0 +1,179 @@
+/*******************************************************************************
+ *     ___                  _   ____  ____
+ *    / _ \ _   _  ___  ___| |_|  _ \| __ )
+ *   | | | | | | |/ _ \/ __| __| | | |  _ \
+ *   | |_| | |_| |  __/\__ \ |_| |_| | |_) |
+ *    \__\_\\__,_|\___||___/\__|____/|____/
+ *
+ *  Copyright (c) 2014-2019 Appsicle
+ *  Copyright (c) 2019-2020 QuestDB
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ *
+ ******************************************************************************/
+
+package io.questdb;
+
+import io.questdb.cairo.*;
+import io.questdb.cairo.security.AllowAllCairoSecurityContext;
+import io.questdb.griffin.SqlCompiler;
+import io.questdb.griffin.SqlException;
+import io.questdb.griffin.SqlExecutionContextImpl;
+import io.questdb.log.Log;
+import io.questdb.log.LogFactory;
+import io.questdb.mp.*;
+import io.questdb.std.*;
+import io.questdb.std.microtime.*;
+import io.questdb.std.str.Path;
+import io.questdb.std.str.StringSink;
+import io.questdb.std.time.MillisecondClock;
+import io.questdb.tasks.TelemetryTask;
+import org.jetbrains.annotations.Nullable;
+
+import java.io.Closeable;
+
+public class TelemetryJob extends SynchronizedJob implements Closeable {
+    private static final Log LOG = LogFactory.getLog(TelemetryJob.class);
+
+    private final CharSequence tableName = "telemetry";
+    private final CharSequence configTableName = "telemetry_config";
+    private final QueueConsumer<TelemetryTask> myConsumer = this::newRowConsumer;
+    private final StringSink idSink = new StringSink();
+
+    private final MillisecondClock clock;
+    private final PropServerConfiguration configuration;
+    private final boolean enabled;
+    private final TableWriter writer;
+    private final TableWriter writerConfig;
+    private final RingQueue<TelemetryTask> queue;
+    private final SCSequence subSeq;
+
+    public TelemetryJob(PropServerConfiguration configuration, CairoEngine engine, @Nullable MessageBus messageBus) throws SqlException, CairoException {
+        final CairoConfiguration cairoConfig = configuration.getCairoConfiguration();
+
+        this.clock = cairoConfig.getMillisecondClock();
+        this.configuration = configuration;
+        this.enabled = configuration.getTelemetryConfiguration().getEnabled();
+        this.queue = messageBus.getTelemetryQueue();
```

Comment:
There is disjointed syntax here. MessageBus is declared nullable and yet the code behaves as if it was declared NotNull